lib/asbind-instance/bind-function.js

```diff
@@ -1,276 +1,180 @@
+function normalizeArrayBufferViewTypeName(typeName) {
+  if (typeName.startsWith("Big")) {
+    // Slice off `Big` as the loader doesn’t have that prefix.
+    typeName = typeName.slice(3);
+  }
+  return typeName;
+}
 
-  // NOTE: Since bound import is being bound to the importObject,
-  // the function this, will become the importObject
-  let functionThis = importObject;
-  const boundImport = function() {
-    const exports = asbindInstance.unboundExports;
+function nop(asbindInstance, value, typeName) {
+  return value;
+}
 
-    // Get the 'this' of the function
-    if (functionThis === importObject) {
-      functionThis = getFunctionFromKeyPath(importObject, [
-        ...importObjectKeyPathToFunction
-      ]);
-    }
+function getString(asbindInstance, value, typeName) {
+  return asbindInstance.exports.__getString(value);
+}
 
-    // Get all arguments as an array
-    const originalArguments = Array.prototype.slice.call(arguments);
+function putString(asbindInstance, value, typeName) {
+  return asbindInstance.exports.__newString(value);
+}
 
-    // Replace all supported reference types,
-    // With their respective references
-    const argumentsWithReplacedRefs = [];
-    originalArguments.forEach((arg, argIndex) => {
-      // Find our supported type, if there is one
-      let supportedType = undefined;
+function getArrayBuffer(asbindInstance, value, typeName) {
+  return asbindInstance.exports.__getArray(value);
+}
 
-      // Check if we already cached the type
-      if (
-        functionThis.shouldCacheTypes &&
-        functionThis.cachedArgTypes[argIndex] &&
-        functionThis.cachedArgTypes[argIndex].type === "ref"
-      ) {
-        supportedType =
-          SUPPORTED_REF_TYPES[functionThis.cachedArgTypes[argIndex].key];
-      } else {
-        Object.keys(SUPPORTED_REF_TYPES).some(key => {
-          if (SUPPORTED_REF_TYPES[key].isTypeFromReference(exports, arg)) {
-            supportedType = SUPPORTED_REF_TYPES[key];
-            if (functionThis.shouldCacheTypes) {
-              functionThis.cachedArgTypes[argIndex] = {
-                type: "ref",
-                key
-              };
-            }
-            return true;
-          }
+function getArrayBufferView(asbindInstance, value, typeName) {
+  return asbindInstance.exports[
+    `__get${normalizeArrayBufferViewTypeName(typeName)}View`
+  ](value);
+}
 
-          return false;
-        });
-      }
+function putArrayBuffer(asbindInstance, value, typeName) {
+  return asbindInstance.exports.__newArray(
+    asbindInstance.exports[
+      `__asbind_${normalizeArrayBufferViewTypeName(typeName)}_ID`
+    ].value,
+    value
+  );
+}
 
-      if (supportedType) {
-        argumentsWithReplacedRefs.push(
-          supportedType.getValueFromRef(exports, arg)
-        );
-      } else {
-        // Assume it was meant to be passed a number
-        argumentsWithReplacedRefs.push(arg);
-      }
-    });
+const ascToJsConverters = new Map([
```

Comment:
OooOoOoOoO this is rad haha! :joy:

Reply:
Glad you like it :D In [another branch](https://github.com/torch2424/as-bind/blob/type-embed-continue/test/tests/custom-type/test.js#L3-L19) I have exposed that map so you can add handlers for your own types (like if you want to polyfill `externref` for example).